### ISSUE TITLE
`source` does not respect $PATH

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#389]
Found this while trying to run `/etc/profile` on my machine: the source builtin does not respect the $PATH variable. From `help source` in bash: 

```
source: source filename [arguments]
    Execute commands from a file in the current shell.
    
    Read and execute commands from FILENAME in the current shell.  The
    entries in $PATH are used to find the directory containing FILENAME.
    If any ARGUMENTS are supplied, they become the positional parameters
    when FILENAME is executed.
    
    Exit Status:
    Returns the status of the last command executed in FILENAME; fails if
    FILENAME cannot be read.
```

Current behavior:
```
(osh) $ . /etc/profile
  . gettext.sh
    ^~~~~~~~~~
/usr/share/im-config/xinputrc.common:15: source 'gettext.sh' failed: No such file or director
```

Expected behavior: sources `/usr/bin/gettext.sh`.

**Comment by andychu** [comment]
Thanks for fixing this!

**Comment by andychu** [comment]
Released with 0.7.pre1: https://www.oilshell.org/release/0.7.pre1/
